localstack-core/localstack/packages/java.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+from typing import List
+
+from localstack.packages import Package
+from localstack.packages.core import ArchiveDownloadAndExtractInstaller
+from localstack.utils.platform import Arch, get_arch, is_linux, is_mac_os
+
+LOG = logging.getLogger(__name__)
+
+# Default version if not specified
+DEFAULT_JAVA_VERSION = "11"
+
+# Supported Java LTS versions mapped with Eclipse Temurin build semvers
+JAVA_VERSIONS = {
+    "8": "8u422-b05",
+    "11": "11.0.24+8",
+    "17": "17.0.12+7",
+    "21": "21.0.4+7",
```

Comment:
With the exact versions, we are loosing the automated updates (by dependabot for the Dockerfile updates, and by Debian repo updates for other versions).
Is there a chance to use the `latest` release link for a specific version to always pick the latest revision / release from temurin?

Reply:
GitHub [allows this](https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases#linking-to-the-latest-release):


> To link directly to a download of your latest release asset that was manually uploaded, the suffix is `/releases/latest/download/asset-name.zip`.

But unfortunately the `asset-name.zip` as published by Adoptium contains a semver. We can't remove the above hardcoded semvers without adding additional logic to query and filter the assets.

Adoptium publishes Java builds as GitHub release entries (eg. https://github.com/adoptium/temurin8-binaries). They don't publish a 'latest stable' release unfortunately. Things are mixed up with beta versions. Their [download page](https://adoptium.net/en-GB/temurin/releases/?version=8) uses an [API endpoint](https://api.adoptium.net/v3/assets/latest/8/hotspot?os=linux&architecture=x64).

In the earlier approach, we had a hardcoded Docker image. I believe this keeps the same behaviour, but I admit automatic bumps would be great